### PR TITLE
Replace kernel injection by FileLocator

### PR DESCRIPTION
**Pull request by zeliard91**
There is no need to inject Kernel in ImageHandling as there is a service to locate resources.

### CODE REVIEW

File: Services/ImageHandling.php [hunk 1]
```diff
@@ -17,16 +17,16 @@ class ImageHandling
     private $cacheDirMode;
     private $container;
     private $handlerClass;
-    private $kernel;
+    private $fileLocator;
     private $throwException;
 
-    public function __construct($cacheDirectory, $cacheDirMode, $handlerClass, ContainerInterface $container, KernelInterface $kernel, $throwException, $fallbackImage)
+    public function __construct($cacheDirectory, $cacheDirMode, $handlerClass, ContainerInterface $container, FileLocator $fileLocator, $throwException, $fallbackImage)
```
Comment (Soullivaneuh) [inline Services/ImageHandling.php]: This is BC break.

**Comment by Soullivaneuh** [comment]
Globally :+1: for the PR but the BC breaking needs to be handled.

**Comment by zeliard91** [comment]
Indeed @Soullivaneuh, I did not thought of BCs, any idea how it could be handled ?

Something like that is BC safe but the purpose of not injecting the kernel is forgotten no ?

```php
public function __construct($cacheDirectory, $cacheDirMode, $handlerClass, ContainerInterface $container, KernelInterface $kernel, $throwException, $fallbackImage)
{
// ...
    $this->fileLocator = $container->get('file_locator');
}
```

**Comment by Soullivaneuh** [comment]
First one, the container have nothing to do here and this ugly things should be removed. (cc @Gregwar :-P).

So no, don't use the container.

What to do:

* Change constructor to remove typehint
* Manually check if argument is an instance of one of these interfaces
* Throw a deprecated warning if the `KernelInterface` is passed
* Use compiler pass for service definition

**Pull request closed without merging**